## 50_gnus-filterhist.diff by Jaakko Kangasharju <ashar@iki.fi>

--- a/elisp/gnus-bonus-el/gnus-filterhist.el	2003-10-17 23:20:53.000000000 +0300
+++ b/elisp/gnus-bonus-el/gnus-filterhist.el	2005-10-13 20:59:18.000000000 +0300
@@ -188,9 +188,9 @@
 				)
 			    )))
 		      (setq nnmail-session-split-history my-session-split-history)
-		  ))))
-	(add-text-properties (point-min) (point-max)
-			     '(face gnus-filterhist-face-1))
+		  ))
+		(add-text-properties (point-min) (point-max)
+				     '(face gnus-filterhist-face-1))))
 	(switch-to-buffer buf)
 	(if gnus-filter-history-popup
 	    (pop-to-buffer "*Filter History*")
